### DISCUSSION TITLE
Link Input Singnature to Output signature in UDF

### DISCUSSION THREAD

**Discussion by momed-sakka** [apache/datafusion#9427] [Ideas]
I would like to define a UDF which extract keys and entries from MapArray into a ListArray.
However creating a UDF requires specification of input types and return_types DataTypes. In this case I need to know the underlying datatypes (keys, values) of the input MapArray beforehand. is there a way to link input signature to the udf output ?

**Comment by alamb** [comment]
In general, DataFusion needs to know the entire datatype during planning and so I don't think you can implement this in general as a user defined function -- you would need to figure out the fields first somehow. 

However, after the newly added  https://github.com/apache/arrow-datafusion/pull/8985 you *could* compute the appropriate type if the argument was a constant value.
